### DISCUSSION TITLE
Tags into filenames and AI integration?

### DISCUSSION THREAD

**Discussion by mastnacek** [TagStudioDev/TagStudio#626] [Ideas]
Hello, can i ask why you dont use renaming files? filename-[tag tag tag].* ? Do you plan integrate AI for tag generating, suggestions? Short explain what i have done.. i have small python script, which generate jsons ( right now ) witch informations from pdf, txt files ( other soon ). This jsons you can use for promt with openai, ollama and so on. Its good to have tags done and ask AI chose from this tags  best for file with this content. Regards Jarda

**Comment by Toby222** [comment]
Filenames are not nearly flexible enough, nor big enough, for proper tagging as is the goal of TagStudio
AI integration is not a goal and will be left as an exercise to the users :⁾

**Reply by mastnacek** [reply]
Not using file names is a shame, it's limiting for one program and I personally don't like that. Assigning a tag in Tagstudio seems very user-unfriendly to me. But I'm giving it a chance.

**Reply by python357-1** [reply]
using file names to store tags would limit tagstudio quite significantly:

- it'd obviously mean that file names are updated, which many users don't want
- tag names could only include letters, numbers, parentheses, underscores, dashes, asterisks, and periods
- it would severely limit the number of tags a file could have
- it would make it effectively impossible to do tag inheritance or composition

i'm sure there are more. i'd suggest you look into the features tagstudio has, it has a lot of stuff you are likely missing doing it your way

**Comment by python357-1** [comment]
are you seriously suggesting that we keep tags as a list of strings in a filename and ask an LLM what the user probably meant when they search for a tag? why would we ever do that?

**Reply by mastnacek** [reply]
You misunderstood me. The approach I'm working on is that, as a user, I define the tags—basic, extended, simply whatever I can come up with myself. I have a bunch of documents: PDFs, DOCs, images. If you can extract information from the file, for example from metadata (author, title, name, etc.), you can compose a prompt and cleverly ask the AI which of my created tags it would select for the given file. Do you understand? Basic bulk assignment of tags. Then the user can add or remove tags as they wish.

**Reply by mastnacek** [reply]
You can make a function in the GUI that the AI ​​will offer you suitable tags, sorted from the most accurate and then the others separately. Everything is possible.

**Reply by python357-1** [reply]
well this seems a lot better than what i thought you were saying, so i apologize for that. however, this is the type of thing that seems out of scope of tag studio. this would probably best be done through an extension when they come around

**Reply by Computerdores** [reply]
To quote the README
> I wouldn't, however, mind using locally hosted models to provide the optional ability for additional searching and tagging methods (especially when it comes to facial recognition) - but this would likely take the form of plugins external to the core program anyway.
